### PR TITLE
Update DeveloperGuide

### LINKED ISSUE #86: Update DG with Design Details
Each member should describe the implementation of at least one enhancement they added
Expected length: 1+ page per person
Description includes
- How the feature is implemented
- Why it is implemented that way
- Alternatives considered

Done:
- [x] Anthony
- [x] Liyan
- [x] Yiqiao
- [x] Zong Jin
- [x] Elgin

### PR DESCRIPTION
**Pull request by 0xteet0**
Changes:
- Included implementation details and considerations for MarkAttendanceCommand
- Included activity diagram for MarkAttendanceCommand
- Updated Use Case for MarkAttendanceCommand

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -185,6 +185,55 @@ In this example, the user enters `list attendance w/1 tg/G2`.
...
+- Not checking for same week's attendance
+
+  > Instead of checking if the attendance for the same week, we could simply add a new attendance record every time the command is invoked. 
+
+  **Pros:**
+    - Simplicity: Implementation would be straightforward without the need for additional checks.
+    - Flexibility: Allows for multiple attendance records within the same week, which might be useful in some scenarios.
+
+  **Cons:**
+    - Redundancy:  Multiple attendance records for the same week can lead to confusion and redundancy.
```
Comment (ElginTZM) [inline docs/DeveloperGuide.md:230]: Minor formatting issue here: Double space

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -415,11 +464,10 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 **MSS**
 
-1.  User decides to mark attendance for a student.
-2.  User enters student name or ID, followed by the attendance status.
-3.  TAvigator marks the attendance of student and displays message for confirmation of attendance.
+1.  User requests to mark attendance for a student  and enters student name or ID, followed by the attendance status and week and reason if any.
```
Comment (ElginTZM) [inline docs/DeveloperGuide.md]: Minor formatting issue here: Double space

File: docs/diagrams/MarkAttendanceCommandActivityDiagram.puml
Comment (ElginTZM) [inline docs/diagrams/MarkAttendanceCommandActivityDiagram.puml:1]: The conditional statements should be in square brackets

**Review by ElginTZM — CHANGES_REQUESTED** [review]
LGTM for the most part, just slight minor formatting issues.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ElginTZM) [inline docs/DeveloperGuide.md]: Minor formatting issue here: Double space

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (0xteet0) [reply docs/DeveloperGuide.md]: Thanks! Updated!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (0xteet0) [reply docs/DeveloperGuide.md]: Thanks! Updated!

File: docs/diagrams/MarkAttendanceCommandActivityDiagram.puml
Comment (0xteet0) [reply docs/diagrams/MarkAttendanceCommandActivityDiagram.puml:1]: Thanks! Updated!

**Review by ElginTZM — APPROVED** [review]
Yep! LGTM 👍

**Pull request merged**